### ISSUE TITLE
An empty wxSpinCtrl is set to its minimum value when losing the focus

### ISSUE THREAD

**Issue by taler21** [wxWidgets/wxWidgets#23563]
### Description

An empty `wxSpinCtrl` or `wxSpinCtrlDouble` is set to its minimum value when it loses input focus, regardless of whether it was emptied programmatically or by the user.

### Expected behaviour

A `wxSpinCtrl` or `wxSpinCtrlDouble` that has been set empty programmatically should remain empty when losing the focus.
It should only be set to its minimum value if the input value is invalid or has been cleared by the user.

### To Reproduce

The issue can be seen in the widgets sample.

Note: With wxGTK it works correct if the value of the spin control was the minimum value before emptying.
So changing the initial values (in step 3) is required to reproduce the issue there.

1. Start the widgets sample.
2. Go to 'Native / Spin' page.
3. Enter the value 5 in both spin controls (wxSpinCtrl and wxSpinCtrlDouble).
4. Click 'Set value' to clear the spin controls programmatically.
5. Set input focus to spin control wxSpinCtrl.
6. Set input focus to spin control wxSpinCtrlDouble.
7. Set input focus to another control.
   See the issue: The spin controls were set to a value instead of staying empty.

Only `wxSpinCtrlDouble` of wxMSW seems to work correct (and remains empty).

### Platform and version information

- wxWidgets version  you use: 3.2.2.1
- wxWidgets port  you use: wxMSW, wxGTK2, wxGTK3
- OS  and its version: Windows 10, openSUSE Leap 42.3

+ GTK version:  2.24.31 and 3.20.10
+ Which GDK  backend is used: X11

**Comment by taler21** [comment]
My use case is an edit dialog that allows multiple objects to be edited together (multiple selection).
If the objects have different values of an attribute (e.g. different stroke width), `wxSpinCtrlDouble` is initialised (programmatically set) to be empty.
Only those attributes of the objects for which the user selects a value in the edit dialog will be changed.
If no value is selected (the control remains empty), the different attributes must remain unchanged.

It's unexpected behavior when a value is selected just because the control gains and loses input focus, for example, when navigating with the tab key.

> why shouldn't the user be able to clear it too then?

Maybe to have consistent behavior with `wxChoice` and `wxComboBox`?
As far as I know, `-1` (no selection) can only be set programmatically, not by the user.

**Comment by vadz** [comment]
Yes, I agree that it's unexpected for the value to change. If we want to guarantee that it always has a valid value, it shouldn't be possible to make it empty at all...

OTOH it risks being difficult to change the native control behaviour in wxGTK.
